site/docs/.vitepress/deps/langs.ts

```diff
@@ -0,0 +1,38 @@
+import { computed } from "vue";
+import { useData } from "vitepress";
+import { ensureStartingSlash } from "./utils.js";
+export function useLangs(
+  { removeCurrent = true, correspondingLink = false } = {},
+) {
+  const { site, localeIndex, page, theme } = useData();
+  const currentLang = computed(() => ({
+    label: site.value.locales[localeIndex.value]?.label,
+    link: site.value.locales[localeIndex.value]?.link ||
+      (localeIndex.value === "root" ? "/" : `/${localeIndex.value}/`),
+  }));
+  const localeLinks = computed(() =>
+    Object.entries(site.value.locales).flatMap(([key, value]) =>
+      removeCurrent && currentLang.value.label === value.label ? [] : {
+        text: value.label,
+        link: normalizeLink(
+          value.link || (key === "root" ? "/" : `/${key}/`),
+          theme.value.i18nRouting !== false && correspondingLink,
+          page.value.relativePath.slice(currentLang.value.link.length - 1),
+          !site.value.cleanUrls,
+        ),
+      }
+    )
+  );
+  return { localeLinks, currentLang };
+}
+// @ts-ignore
+function normalizeLink(link, addPath, path, addExt) {
```

Comment:
Please add a comment explaining what normalisation is done and why

Reply:
The files in the 'deps' folder are from Vitepress. Since I couldn't import them directly, I duplicated the files into that directory and imported them from there.

Reply:
Wait so you didn't write the code in the deps folder? That sounds a lot like we should rather find a way to import the functionality, not copy the build output into this repo.

Reply:
Maybe I need to take a second look to understand more about this

Reply:
Yes, please. :')